### ISSUE TITLE
[Due for Payment 2026-04-10][$150] [A11y] Screen Reader: Sign in: The language dropdown is announced as disabled and cannot be activated

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77542]
## Action Performed:
Prerequisite:
- the user is not signed in

1. Using Android TalkBack, open the Expensify app
2. Swipe to the language dropdown button in the footer
3. Note how the control is announced.
4. Try to activate the control by double tapping

## Expected Result:
The control should be announced without a disabled state and the screen reader user should be able to activate it by double tapping.

## Actual Result:
When a user attempts to focus on the dropdown, TalkBack announces the component's name and state: "[Language Name], button disabled." The component also does not respond to the double-tap gesture, rendering the language selection inaccessible.

Note: the button can be activated without enabled TalkBack.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Remove the disabled state and ensure the control can be activated with an enabled screen reader.
https://reactnative.dev/docs/accessibility#accessibilitystate

## Environment:
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)

## Screenshots/Videos

![Image](https://github.com/user-attachments/assets/1492ff65-7e9c-4209-8aa6-ba84da363f8d)

https://github.com/user-attachments/assets/ad83a619-a015-4448-8b13-2d77e26fe0ae

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005441
**Issue reported by:** NA
**Slack conversation:** NA
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022009685274860295821</li>
        <li>Upwork Job ID: 2009685274860295821</li>
        <li>Last Price Increase: 2026-04-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by rushatgabhane** [comment]
the correct amount for this issue is $75

**Comment by TaduJR** [comment]
Hello @inimaga 

Rasing the PR today. I was testing on all platforms not only on the reported one that why it took time

**Comment by TaduJR** [comment]
Update: 

So there debugging mistake when writing the Proposal. I was testing with a wrong gestures that bypasses android accessibility. Double Tap, Hold and then release works but not double tap.

I was investigating a lot, phew it was exhausting, and the fix should be in upstream. I already made a PR on expensify native picker select.

BTW this is a known issue on the original upstream here `https://github.com/lawnstarter/react-native-picker-select/issues/447`

Will make a for the original upstream lib PR [lawnstarter](https://github.com/lawnstarter)[react-native-picker-select](https://github.com/lawnstarter/react-native-picker-select) as well to fix the issue for all

**Comment by inimaga** [comment]
Same update as [this](https://github.com/Expensify/App/issues/77542#issuecomment-3797077951).

**Comment by TaduJR** [comment]
Hi @inimaga 

I have some questions about [react-native-picker-select](https://github.com/Expensify/react-native-picker-select) fork. The README says 

> This fork should be deprecated once https://github.com/lawnstarter/react-native-picker-select/pull/464 has been merged into the upstream library.

And it appears that PR is merged, so is there any reason that we still should be using this fork. Also the fork is not updated it's way behind the original lib.

**Comment by inimaga** [comment]
Hmmm, good question. I'ld lean towards deprecating our fork and using the original library. Raised the question internally [here](https://expensify.slack.com/archives/C03TQ48KC/p1770140855164069) to find out if there is a reason it hasn't been deprecated yet.

**Comment by inimaga** [comment]
No, not intentional. The closing seems to have happened automatically on me merging [this PR](https://github.com/Expensify/react-native-picker-select/pull/19).

**Comment by rushatgabhane** [comment]
Hey @TaduJR https://github.com/Expensify/react-native-picker-select/pull/22 our upgrade doesn't include this PR right?

**Comment by rushatgabhane** [comment]
@MelvinBot create an upgrade PR please for latest commit for react-native-picker-select

**Comment by eVoloshchak** [comment]
# BugZero Checklist:

- [x] **[Contributor]** The offending PR and associated issue have been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to the comment on the Issue: N/A, the accesibility labels were not added in the first place (we're using a fork of a third-party library)

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: regression wasn't critical

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.


- [x] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue: https://github.com/Expensify/Expensify/issues/623479

## Regression Test Proposal
### Precondition:


- You must be logged out.

### Test:


Test 1: Android TalkBack & iOS VoiceOver - Language picker announcement and activation

1. Using Android TalkBack, open the Expensify app
2. Swipe right to focus the 'Language' picker at the bottom of the page
3. Verify TalkBack announces Collapsed, Language, combobox (not "button disabled")
4. Double-tap to activate
5. Verify the language picker opens and a language can be selected

Test 2: iOS VoiceOver - Done button role

1. Using iOS VoiceOver, open the Expensify app
2. Swipe to the 'Language' dropdown button and double-tap to activate
3. Swipe to the 'Done' button in the picker view
4. Verify that the Done button is announced with a button role

Do we agree 👍 or 👎



<details><summary>Zapier Logs</summary>
Run ID: <a href="https://zapier.com/editor/203475862/run/00040eee-ffd6-addc-d3a5-cd220d5621b4/run-details">00040eee-ffd6-addc-d3a5-cd220d5621b4</a>
</details>

<details><summary>Zapier Logs</summary>
Run ID: <a href="https://zapier.com/editor/203475862/run/00040eee-1f7c-a460-609f-715448565cd4/run-details">00040eee-1f7c-a460-609f-715448565cd4</a>
</details>

**Comment by mallenexpensify** [comment]
@rushatgabhane assigned and made ya owner for 👀 to see if you think this warrents a price increase cuz of the upstream fix.  If not, please close.  If so, please propose your recommended amount to increase.

**Comment by mallenexpensify** [comment]
^ Test case created

**Comment by rushatgabhane** [comment]
> see if you think this warrents a price increase cuz of the upstream fix

@mallenexpensify maybe?? normally it would, but we own react-native-picker-select so maybe the current price is fair.
@inimaga wdyt?

**Comment by inimaga** [comment]
>see if you think this warrents a price increase cuz of the upstream fix
>
> @mallenexpensify maybe?? normally it would, but we own react-native-picker-select so maybe the current price is fair.
> @inimaga wdyt?


I think we should increase the price for this by $75. Although we own `react-native-picker-select` fork, the work went beyond a normal App fix, as it required fixing our fork, merging upstream, handling follow-ups, and updating App to consume the fork. 

So perhaps $325? @mallenexpensify

**Comment by mallenexpensify** [comment]
Thanks @inimaga .  I missed [this](https://github.com/Expensify/App/issues/77542#issuecomment-3730342639) earlier
> the correct amount for this issue is $75

When Rushat [re-added ](https://github.com/Expensify/App/issues/77542#event-21952872538)`External` it bumped the price to $250. 

Since the summary states $250 above, I propose we leave it at that.  Anyone disagree?

**Comment by TaduJR** [comment]
I think we went beyond the scope here https://github.com/Expensify/react-native-picker-select/pull/21#pullrequestreview-3918271206

From my side merging with main many conflicts and fixing the failing workflows for main was a little tough?

**Comment by inimaga** [comment]
> Since the summary states $250 above, I propose we leave it at that. Anyone disagree?

If its already upped, then I agree.

> From my side merging with main many conflicts and fixing the failing workflows for main was a little tough?

@TaduJR But this added potential complexity was why it was upped in the first place. And fixing the failing workflow is part of that PR.

**Comment by TaduJR** [comment]
Nice. So what's left here?

**Comment by inimaga** [comment]
Over to @mallenexpensify

**Comment by mallenexpensify** [comment]
Closing, [summary should be accurate above.](https://github.com/Expensify/App/issues/77542#issuecomment-4214716742) Thx

**Comment by JmillsExpensify** [comment]
$250 approved from @eVoloshchak
